optuna/study.py

```diff
@@ -291,6 +292,13 @@ def optimize(
                     _iter = iter(int, 1)
 
                 with Parallel(n_jobs=n_jobs, prefer="threads") as parallel:
+                    if not isinstance(parallel._backend, joblib.parallel.ThreadingBackend) and \
+                       isinstance(self._storage, storages.InMemoryStorage):
+                        msg = '`InMemoryStorage` cannot be shared by multiple processes. Please' \
```

Comment:
I'm worried that "InMemoryStorage" and "RDBStorage" in this message seem rather developer terminology, and users may not be able to understand the meaning of the message well.
How about rewriting the message to what that doesn't refer to the class names (e.g., "The default in-memory storage cannot be shared by multiple processes. Please use RDB storage by specifying `storage` argument to the study when a joblib backend that distributes tasks to multi processes is selected").

Reply:
Thank you for your comment and your alternative message.

> "InMemoryStorage" and "RDBStorage" in this message seem rather developer terminology, and users may not be able to understand the meaning of the message well.

That makes sense. But, I still think the technical terms (especially for the case of `RDBStorage`) could provide crews to search for detailed descriptions. So, I updated the message as follows:

"The default storage cannot be shared by multiple processes. Please use an RDB (`RDBStorage`)  when you use joblib for multi-processing. The usage of `RDBStorage` can be found in https://optuna.readthedocs.io/en/stable/tutorial/rdb.html."

What do you think?

Reply:
That makes sense. Your new message looks better than my proposed one.

Reply:
Thank you. I updated the comment (https://github.com/optuna/optuna/pull/780/commits/eb825b38f2b596e32d05d05f8c51829f6a0693a8).
Please take another look.